lib/resty/radixtree.lua

```diff
@@ -763,8 +762,29 @@ local function match_route_opts(route, opts, args)
         end
     end
 
+    local opts_vars = opts.vars or ngx_var
+    local param_matches, param_names
+    if route.param then
+        local matched
+        matched, param_matches, param_names = match_route_params(path, route, opts)
+        if not matched then
+            return false
+        end
+
+        -- Allow vars expr or filter_fun to use `uri_param_<name>`
+        if (route.vars or route.filter_fun) and param_names and param_matches then
+            opts_vars = clone_tab(opts_vars)
```

Comment:
this may wrong, you can take a look at this: https://github.com/apache/apisix/blob/master/apisix/core/ctx.lua#L318

this solution seems a little complex

Reply:
Nice spot @membphis 

I added a test case and a tweak to use the `__index` metamethod in https://github.com/api7/lua-resty-radixtree/pull/142/commits/22f00104216634be9420fd437d9b246fb9e48be2

FYI, I tried using metamethods before in https://github.com/api7/lua-resty-radixtree/pull/142/commits/2b9f67dcad7b51624e7b6a3d625ef8bc794ae166 but they where to slow.

Reply:
Also what do you think is complex here and what would you suggest to do?

Reply:
lgtm